libraries/splunk_app.rb

```diff
@@ -278,6 +297,19 @@ def manage_metaconf
         manage_file(new_resource.perms_file, permissions)
       end
 
+      def download_file(file_path, url)
+        download = Chef::Resource::RemoteFile.new(file_path, run_context)
+        download.source(url)
+        download.run_action(:create)
+        download.backup(false)
```

Comment:
I think this needs to be before the previous line to have any effect... run_action actually invokes the action to be performed, and then you setting an attribute afterwards.

See: https://www.chef.io/blog/2013/09/04/demystifying-common-idioms-in-chef-recipes/